cmd/kots/cli/deploy.go

```diff
@@ -0,0 +1,363 @@
+		return errors.Wrap(err, "failed to create license sync http request")
+	}
+	newRequest.Header.Add("Authorization", authSlug)
+	newRequest.Header.Add("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(newRequest)
+	if err != nil {
+		return errors.Wrap(err, "failed to execute license sync http request")
+	}
+	defer resp.Body.Close()
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "failed to read license sync server response")
+	}
+
+	response := struct {
+		Success bool   `json:"success"`
+		Error   string `json:"error,omitempty"`
+		Synced  bool   `json:"synced"`
+	}{}
+	_ = json.Unmarshal(respBody, &response)
+
+	if resp.StatusCode != http.StatusOK {
+		if response.Error != "" {
+			return errors.Errorf("license sync failed: %s", response.Error)
+		}
+		return errors.Errorf("license sync failed with status code %d", resp.StatusCode)
+	}
+
+	if response.Synced {
+		log.ActionWithoutSpinner("License synced successfully")
+	} else {
+		log.ActionWithoutSpinner("License already up to date")
+	}
+
+	return nil
+}
+
+func handleUpstreamUpdate(v *viper.Viper, appSlug string, localPort int, authSlug string, log *logger.CLILogger) error {
+	log.ActionWithoutSpinner("Processing upstream update...")
+
+	airgapBundle := v.GetString("airgap-bundle")
+	isAirgap := airgapBundle != ""
+
+	var requestBody io.Reader
+	var contentType string
+
+	if isAirgap {
+		// Create multipart form data for airgap bundle using shared function
+		var err error
+		requestBody, contentType, err = upstream.CreateAirgapMultipartRequest(airgapBundle)
+		if err != nil {
+			return err
+		}
+	} else {
+		requestBody = strings.NewReader("{}")
+		contentType = "application/json"
+	}
+
+	urlVals := url.Values{}
+	if v.GetBool("skip-preflights") {
+		urlVals.Set("skipPreflights", "true")
+	}
+	if channelID := v.GetString("channel-id"); channelID != "" {
+		urlVals.Set("channelId", channelID)
+	}
+	if channelSequence := v.GetInt64("channel-sequence"); channelSequence != 0 {
+		urlVals.Set("channelSequence", strconv.FormatInt(channelSequence, 10))
+	}
+
+	upstreamURL := fmt.Sprintf("http://localhost:%d/api/v1/app/%s/upstream/update?%s", localPort, url.QueryEscape(appSlug), urlVals.Encode())
+
+	newRequest, err := http.NewRequest("POST", upstreamURL, requestBody)
+	if err != nil {
+		return errors.Wrap(err, "failed to create upstream update http request")
+	}
+	newRequest.Header.Add("Authorization", authSlug)
+	newRequest.Header.Add("Content-Type", contentType)
+
+	resp, err := http.DefaultClient.Do(newRequest)
+	if err != nil {
+		return errors.Wrap(err, "failed to execute upstream update http request")
+	}
+	defer resp.Body.Close()
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "failed to read upstream update server response")
+	}
+
+	response := struct {
+		Success bool   `json:"success"`
+		Error   string `json:"error,omitempty"`
+	}{}
+	_ = json.Unmarshal(respBody, &response)
```

Comment:
can you handle this error always or at least for the successful case (if resp.StatusCode == http.StatusOK)